### PR TITLE
feat(pci): log to customer - live tail

### PR DESCRIPTION
**Pull request by dectotam**
| Question         | Answer
| ---------------- | ---
| Branch?          | feat/log-to-customer
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Feat MANAGER-12634
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description

init of the new log to customer component : logs poll, search, pause, clear session, expand

### CODE REVIEW

File: packages/components/ng-log-to-customer/src/controller.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    LogToCustomer,
+    $http,
+    $interval,
+    $timeout,
+    $sce,
+    $transclude,
+  ) {
+    this.$translate = $translate;
+    this.LogToCustomer = LogToCustomer;
+    this.$http = $http;
```
Comment (darsene) [inline packages/components/ng-log-to-customer/src/controller.js:18]:
```suggestion
```
Unused

File: packages/components/ng-log-to-customer/src/controller.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
+import flatten from 'lodash/flatten';
+import uniqBy from 'lodash/uniqBy.js';
+import { DISPLAY_RULES } from './constants';
+
+export default class LogToCustomerCtrl {
+  /* @ngInject */
+  constructor(
+    $translate,
+    LogToCustomer,
+    $http,
```
Comment (darsene) [inline packages/components/ng-log-to-customer/src/controller.js:10]:
```suggestion
```
Unused

File: packages/components/ng-log-to-customer/src/controller.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  }
+
+  setUrlSource() {
+    return this.LogToCustomer.getLogSourceUrl(this.source)
+      .then((data) => {
+        this.url = data.url;
+
+        this.setUrlTimeout = this.$timeout(() => {
+          this.setUrlSource();
+        }, Date.parse(data.expirationDate) - new Date().getTime() - 300);
```
Comment (darsene) [inline packages/components/ng-log-to-customer/src/controller.js:162]: Set timeout value in a variable or a function to render it more readable and easier to understand

`const expirationConnectionTimeout = Date.parse(data.expirationDate) - new Date().getTime() - 300)`

**Review by antonymarion — APPROVED** [review]

File: packages/components/ng-log-to-customer/src/translations/Messages_fr_FR.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "clear_session": "Effacer la session",
+  "search_placeholder": "Rechercher un élément ?",
+  "list_no_logs": "En attente de logs...",
+  "error_get_url": "Une erreur a été rencontrée lors de la récupération des logs"
```
Comment (darsene) [inline packages/components/ng-log-to-customer/src/translations/Messages_fr_FR.json:5]: `Une erreur est survenue` is more common
```suggestion
  "error_get_url": "Une erreur est survenue lors de la récupération des logs"
```

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/audit-logs/audit-logs.constant.js [hunk 5]
```diff
@@ -4,14 +4,22 @@ export const LOGS_INFO = {
     'https://docs.ovh.com/gb/en/logs-data-platform/kubernetes-fluent-bit/',
 };
 
-export const HTTP_STATUS = {
-  MAX_SUCCESS_CODE: 299,
-  MIN_REDIRECTION_CODE: 300,
-  MAX_ERROR_CODE: 499,
-  MIN_INTERNAL_ERROR_CODE: 500,
```
Comment (darsene) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/audit-logs/audit-logs.constant.js:11]: no legacy usage of those constants ?

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/audit-logs/audit-logs.html [hunk 6]
```diff
@@ -9,74 +9,6 @@
...
-            <br />
-        </div>
-
-        <div
-            data-translate="kube_logs_list_no_logs"
-            data-ng-if="$ctrl.logger.logs.length === 0"
-        ></div>
-    </tail-logs>
+    <log-to-customer source="$ctrl.logSrcUrl" log-keys="$ctrl.LOG_KEYS">
+    </log-to-customer>
```
Comment (darsene) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/audit-logs/audit-logs.html:13]: no right tile for now ?

Comment (dectotam) [reply packages/manager/modules/pci/src/projects/project/kubernetes/details/audit-logs/audit-logs.html:13]: No, but it is planned for the next task. Still, I prepared the code with ng-transclude and according CSS already

File: packages/components/ng-log-to-customer/src/controller.js [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+    this.url = null;
+    this.DISPLAY_RULES = DISPLAY_RULES;
+    this.errorMessage = null;
+    this.isRightTileSlotFilled = this.$transclude.isSlotFilled('tile');
+    this.fullScreen = !this.isRightTileSlotFilled;
+
+    this.keys = Object.keys(this.logKeys);
+    this.setUrlTimeout = null;
+
+    this.setUrlSource().then(() => this.startLog());
```
Comment (darsene) [inline packages/components/ng-log-to-customer/src/controller.js:37]: `.then()` will be called even though you have an error as you catch it. You should avoid starting log if you don't have any `url`

File: packages/components/ng-log-to-customer/src/controller.js [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+    this.interval = null;
+  }
+
+  clearSession() {
+    this.logs.length = 0;
+  }
+
+  startLog() {
+    this.interval = this.$interval(() => {
+      this.LogToCustomer.getLogs(this.url).then((data) => {
```
Comment (darsene) [inline packages/components/ng-log-to-customer/src/controller.js:104]: You should add a `.catch` block in case of error and display an error message

**Pull request merged**